taichi/program/program.cpp

```diff
@@ -476,6 +477,24 @@ Kernel &Program::get_snode_writer(SNode *snode) {
 
 void Program::finalize() {
   TI_TRACE("Program finalizing...");
+  if (config.print_benchmark_stat) {
+    char *current_test = std::getenv("PYTEST_CURRENT_TEST");
+    if (current_test != nullptr) {
+      std::string file_name = current_test;
+      auto last_colon_pos = file_name.find_last_of(':');
+      TI_ASSERT(last_colon_pos != file_name.npos);
+      file_name = file_name.substr(last_colon_pos + 1);
+      auto first_space_pos = file_name.find_first_of(' ');
+      TI_ASSERT(first_space_pos != file_name.npos);
+      file_name = file_name.substr(0, first_space_pos);
+      file_name += ".log";
+      std::ofstream ofs(file_name);
```

Comment:
What if there are two tests with the same name in different files?

Reply:
Now I also use the name of the test file.

Reply:
Sounds good.